File: backend/src/main/java/com/zlinks/package_system/entity/Notification.java
package com.zlinks.package_system.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("notification")
public class Notification extends BaseEntity {

    private String title;
    private String content;
    private String module;
    private Long targetId;
    private String targetType;
    private Long senderId;
    private String receiverIds;
    private String receiverType;
    private Integer isPinned;
    private String status;
}

File: frontend/src/types/notification.ts
export interface Notification {
  id: number
  title: string
  content: string
  module: string
  targetId: number | null
  targetType: string
  senderId: number | null
  senderName: string
  receiverIds: string
  receiverNames: string
  receiverType: string
  isPinned: number
  status: string
  createBy: string
  createTime: string
  updateBy: string
  updateTime: string
  remark: string
}

export interface NotificationListParams {
  current?: number
  size?: number
  title?: string
  module?: string
  status?: string
}

export interface NotificationRequest {
  title: string
  content: string
  module?: string
  targetId?: number
  targetType?: string
  receiverIds?: number[]
  receiverType?: string
  isPinned?: number
  status?: string
}
